Modules/mmapmodule.c

```diff
@@ -255,6 +258,120 @@ do {                                                                    \
 } while (0)
 #endif /* UNIX */
 
+// copied from bytesobject.c
+#define CHARACTERS _Py_SINGLETON(bytes_characters)
+#define CHARACTER(ch) \
+     ((PyBytesObject *)&(CHARACTERS[ch]));
+
+#if defined(MS_WIN32) && !defined(DONT_USE_SEH)
+static DWORD
+filter_page_exception(EXCEPTION_POINTERS *ptrs, EXCEPTION_RECORD *record)
+{
+    *record = *ptrs->ExceptionRecord;
+    if (ptrs->ExceptionRecord->ExceptionCode == EXCEPTION_IN_PAGE_ERROR) {
+        return EXCEPTION_EXECUTE_HANDLER;
+    }
+    return EXCEPTION_CONTINUE_SEARCH;
+}
+#endif
+
+int
+safe_memcpy(void *restrict dest, const void *restrict src, size_t count) {
```

Comment:
This one can be defined using the macro as well, right?

I'm reading this section from the POV of someone who develops on Linux, trying to figure out where to add the code to make Linux work as well. If that can be made obvious (perhaps with a `// TODO: ...` comment), it's more likely we'll get someone come and add it.

Reply:
Yes it can be defined using the macro too. So far I kept it like this because I had this optional check for zero length inputs. This check is either totally unnecessary though or doesn't hurt to be part of the non-windows implementation.